kratos/processes/apply_periodic_boundary_condition_process.cpp

```diff
@@ -0,0 +1,357 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Aditya Ghantasala
+//
+//
+
+// System includes
+#include <string>
+#include <iostream>
+#include <chrono>
+
+// External includes
+
+
+// Project includes
+#include "utilities/openmp_utils.h"
+#include "processes/apply_periodic_boundary_condition_process.h"
+#include "utilities/binbased_fast_point_locator_conditions.h"
+
+
+namespace Kratos
+{
+
+    ApplyPeriodicConditionProcess::ApplyPeriodicConditionProcess(ModelPart &rMasterModelPart, ModelPart &rSlaveModelPart,
+                                  Parameters rParameters) : Process(Flags()), mrMasterModelPart(rMasterModelPart),
+                                   mrSlaveModelPart(rSlaveModelPart), mParameters(rParameters)
+    {
+        Parameters default_parameters(R"(
+                                            {
+                                            "variable_names":[],
+                                            "center":[0,0,0],
+                                            "axis_of_rotation":[0.0,0.0,0.0],
+                                            "angle":0.0,
+                                            "dir_of_translation":[0.0,0.0,0.0],
+                                            "magnitude":0.0
+                                            }  )");
+
+        // Initializing
+        mParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+        mCenterOfRotation.push_back(mParameters["center"][0].GetDouble());
+        mCenterOfRotation.push_back(mParameters["center"][1].GetDouble());
+        mCenterOfRotation.push_back(mParameters["center"][2].GetDouble());
+
+        mAxisOfRotationVector.push_back(mParameters["axis_of_rotation"][0].GetDouble());
+        mAxisOfRotationVector.push_back(mParameters["axis_of_rotation"][1].GetDouble());
+        mAxisOfRotationVector.push_back(mParameters["axis_of_rotation"][2].GetDouble());
+
+        mModulus = mParameters["magnitude"].GetDouble();
+        mDirOfTranslation.push_back(mParameters["dir_of_translation"][0].GetDouble());
+        mDirOfTranslation.push_back(mParameters["dir_of_translation"][1].GetDouble());
+        mDirOfTranslation.push_back(mParameters["dir_of_translation"][2].GetDouble());
+
+        mTheta = mParameters["angle"].GetDouble() * 2 * 3.1416 / 360.0;
```

Comment:
`3.1416` => `Globals::Pi`

also the name should indicate thaz you expect the input in degree => e.g. `angle_degree`